CsharpExtras/Extensions/CollectionExtension.cs

```diff
@@ -8,7 +8,7 @@ namespace CsharpExtras.Extensions
 {
     public static class CollectionExtension
     {
-        public static bool TryGetSingleton<TValueType>(this ICollection<TValueType> collection, out TValueType singleton)
+        public static bool TryGetSingleton<TValueType>(this ICollection<TValueType> collection, out TValueType? singleton) where TValueType : class

```

Comment:
I advise to rename ``TValueType`` to something like ``TVal``. "Value type" in C# has a very specific meaning and we don't want to confuse people. I'm aware that this wasn't a change you made in this PR but I'm just noticing it now.